db/hints/resource_manager.cc

```diff
@@ -189,7 +188,7 @@ void space_watchdog::on_timer() {
 
         // Adjust the quota to take into account the space we guarantee to every end point manager
         size_t adjusted_quota = 0;
-        size_t delta = boost::accumulate(per_device_limits.managers, 0, [] (size_t sum, manager& shard_manager) {
+        size_t delta = std::ranges::fold_left(per_device_limits.managers, 0, [] (size_t sum, manager& shard_manager) {
```

Comment:
I'm confused, doesn't std::accumulate do exactly this (and is more similar in name to boost::accumulate)? How is fold_left different?

Reply:
yes, it does. in this use case, the only differences between them is that `std::accumulate()` needs two iterators to define a range, while `std::ranges::fold_left()` accepts a range. the former needs more typing, and could be a little bit distracting for human readers -- one needs to parse more tokens to understand the code.